packages/node-core/src/configure/ProjectUpgrade.service.ts

```diff
@@ -107,16 +111,18 @@ export class ProjectUpgradeSevice<P extends ISubqueryProject = ISubqueryProject>
   }
 
   async init(
-    metadata: CacheMetadataModel,
+    storeCache: StoreCacheService,
+    schemaMigrationService: SchemaMigrationService,
```

Comment:
I think now it would make more sense to construct this class in here rather than passing it as an argument. StoreCache should have the relevant information